File: 프로그래머스/lv3/43162. 네트워크/네트워크.cpp
#include <vector>
#include <stack>

using namespace std;


int solution(int n, vector<vector<int>> computers) {
    int answer = 0;
    vector<bool> visited(n, false);
    
    for(int i = 0; i < n; i++){
        if(visited[i])
            continue;
        else {
            visited[i] = true;
            stack<int> s;
            s.push(i);
            
            while(!s.empty()){
                int cur = s.top();
                s.pop();
                visited[cur] = true;
                
                for(int i = 0; i < n; i++){
                    if(visited[i])
                        continue;
                    if(computers[cur][i] == 0)
                        continue;
                    visited[i] = true;
                    s.push(i);
                }
            }
            answer++;
        }
    }
      
    return answer;
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
def solution(n, computers):
    answer = 0
    visited = [False] * (n)

    for i in range(n):
        if not visited[i]:
            visited[i] = True
            stack = [i]

            while len(stack) > 0:
                cur = stack.pop()
                visited[cur] = True

                for j in range(n):
                    if not visited[j] and computers[cur][j] == 1:
                        visited[j] = True
                        stack.append(j)

            answer += 1

    return answer